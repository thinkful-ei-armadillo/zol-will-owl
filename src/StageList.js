import React from 'react';
import Stage from './Stage';
import './stagelist.css';


function StageList(props){
    const onStageParticipants = props.participants.filter(participant => participant.onStage);

    return (
        <div className="stageList">
            {onStageParticipants.map(participant => 
                <Stage 
                    key = {participant.id}
                    name = {participant.name}
                    avatar = {participant.avatar}
                />
            )}
        </div>
    )
}

export default StageList