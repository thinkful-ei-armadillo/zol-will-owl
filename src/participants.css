img {
    width: 40px;
    border-radius: 50%;
    border: 1px solid black;
    background: gray;
}

.participantInfo {
    display: inline-block;
    padding-left: 15px;
}


background: rgb(187, 187, 187);
div.participant {
    margin-top: 20px;
    margin-bottom: 20px; 
}

@media ( min-width : 600px ){
  .participantList {
    width: 30%;
    height: 100vh;
  }
}

