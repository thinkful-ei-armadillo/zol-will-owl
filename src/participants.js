import React from 'react';

function User(props) {

    return (
        <div className='user'>
            
        </div>
    )
}


function Participants(props) {

    return (
    // id
    // name
    // avatar
    // session: true or false
    )

}


export default Participants;