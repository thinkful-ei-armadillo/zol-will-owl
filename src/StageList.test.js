import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import StageList from './StageList';
import participants from './STORE';


describe('Messages component', () => {
  it('renders chatlist without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<StageList participants={participants}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders chatlist', () => {
    const tree = renderer
      .create(<StageList participants={participants}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});