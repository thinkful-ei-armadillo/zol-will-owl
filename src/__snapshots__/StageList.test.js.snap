// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages component renders chatlist 1`] = `
<div
  className="stage-list"
>
  <div
    className="stage"
  >
    <div>
      <span
        className="name"
      >
        Koren Templeton
      </span>
    </div>
    <img
      alt="Koren Templeton"
      className="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="stage"
  >
    <div>
      <span
        className="name"
      >
        Ashla Attwool
      </span>
    </div>
    <img
      alt="Ashla Attwool"
      className="avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
  </div>
</div>
`;
